org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -703,4 +709,44 @@ public void projectsBuildFinished(IProgressMonitor monitor) {
 	public boolean isBuildFinished() {
 		return buildFinished;
 	}
+
+	@Override
+	public void checkIndexes() {
+		Job job = new Job(("Check Indexes")) {
+
+			@Override
+			protected IStatus run(IProgressMonitor monitor) {
+				IJavaProject[] javaProjects = ProjectUtils.getJavaProjects();
+				if (javaProjects == null || javaProjects.length <= 0) {
+					return Status.OK_STATUS;
+				}
+				SearchEngine engine = new SearchEngine();
```

Comment:
Could we reuse https://github.com/eclipse-jdtls/eclipse.jdt.ls/blob/e90ec8f61d3b7918e4da8fc38a2d17eb4ff8893a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JobHelpers.java#L250 ?

Reply:
>  Does the search term matter that much ?

Yes, it does. We search for a JDK class.

Reply:
This only happens for `JDK` classes ? Why does that "index" have problems and not others ?

Reply:
I suppose JDT builds JDK indexes in a specific way. I will check.

Reply:
@snjeza , did you find a way to reproduce this more consistently ? Maybe investigating why the JDK index is specifically incorrect would indicate how this happens ?

Reply:
> did you find a way to reproduce the issue more consistently ?

No, I didn't. I use the steps described at https://github.com/redhat-developer/vscode-java/issues/3797#issuecomment-2395280866

> Maybe investigating why the JDK index is specifically incorrect would indicate how this happens ?

I'm trying to find the issue.
However, since JDT UI uses `JavaCore.initializeAfterLoad()' Java LS should also use it.